community-operators/triggermesh/triggermesh.v0.0.1.clusterserviceversion.yaml

```diff
@@ -0,0 +1,175 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: '[{"apiVersion":"app.triggermesh.io/v1alpha1","kind":"TriggerMesh","metadata":{"name":"example-triggermesh"},"spec":{"backend":{"baseHost":"app-test.tm.demo.vshn.net","route":{"annotations":{"kubernetes.io/tls-acme":"true"},"enabled":true}},"frontend":{"authApiId":"\u003cREDACTED\u003e","baseHost":"cloud-test.tm.demo.vshn.net","route":{"annotations":{"kubernetes.io/tls-acme":"true"},"enabled":true},"sentryDSN":"https://\u003cREDACTED\u003e@sentry.io/"},"k8sAPI":{"CA":"-----BEGIN
+      CERTIFICATE-----\nMIIC6jCCAdKgAwIBAgIBATANBgkqhkiG9w0BAQsFADAmMSQwIgYDVQQDDBtvcGVu\nc2hpZnQtc2lnbmVyQDE1NTQzMDQ3MDIwHhcNMTkwNDAzMTUxODIyWhcNMjQwNDAx\nMTUxODIzWjAmMSQwIgYDVQQDDBtvcGVuc2hpZnQtc2lnbmVyQDE1NTQzMDQ3MDIw\nggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDd/NHyQyCg6a65VywMe3TT\nmuam6XK3gd3iB9vPyTXnzYTx2IEc9vRG9kYmUwu5XE7ZsOXKNaT6Xez1FPhX1FfL\nfKL9MC4n86sFRiT1zE2DfW6ycL6rJ2XaxmAe3HSw5q7FrgH9EWEC3lLPkO4Xygn+\n0T8G3sdBNbPIK8LGLjb1RBeM8Tif1AUABehCY9eb9suY+zzU1/RJ9hY5Y1X4yI4l\n5yotE3KzwL/aphiBiQGXajQLzPg7tFlE5qUuHRgo07Cqsyt9m89gVf8s3s3hfRPe\nttjA5dBnkjfaI8POJp71Nof2rXy2zaDG7QBkILrNozN1PJFH9f6+uCFKkJ8eg/Br\nAgMBAAGjIzAhMA4GA1UdDwEB/wQEAwICpDAPBgNVHRMBAf8EBTADAQH/MA0GCSqG\nSIb3DQEBCwUAA4IBAQCkSB7BiLxBsFIm/JkyoKBupB4MB4GIqNTs+Pxwus0Z2S0U\nfW1QKh745nvNmHkXIaRyKh8dkIqIQDmUFo9FlxiS5zXxAe3F6OYmxAJ3sE24oA1m\nnoyDQ/TlaxLxLFfVo8q3fc/dIYlvyTMeFjoNb22ZClAHgXgwsxJIKhLl7J1Nux4N\nq7nCj1tBff7WGNkzN0Wj7BTMLfCcUpTYXxtRq9gmaXXxWbFLV+B4ViYg1VNJ8VSZ\n/DYgkGOKzO7Y9DFq90gVpat7+XNOGHDZtPP+IiTQZwyONxKT/b1UHJm2Q7iD23Yt\nk302uCcbAC6JcP7+lCmH/X3ZgevgeAe715kwj8Lj\n-----END
+      CERTIFICATE-----\n"},"secrets":{"auth0ClientId":"\u003cREDACTED\u003e","auth0ClientSecret":"\u003cREDACTED\u003e","gitHookSecret":"\u003cREDACTED\u003e"}}}]'
+    capabilities: Basic Install
+    containerImage: gcr.io/triggermesh/operator:v0.0.1
+    repository: https://github.com/triggermesh/operator
+    createdAt: 2019-04-11
+    support: TriggerMesh Inc.
+    description: |-
+      A serverless management platform that runs on Knative. TriggerMesh provides continuous delivery of serverless functions, build, auto-scaling and eventing capabilities.
+    categories: Developer Tools, Streaming & Messaging, Integration & Delivery 
+  name: triggermesh.v0.0.1
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions:
+    owned:
+    - kind: TriggerMesh
+      name: triggermeshes.app.triggermesh.io
+      version: v1alpha1
+      displayName: Running TriggerMesh instance
+      description: Configures TriggerMesh with Auth0 authentication, GitHub secret 
+  description: |
+    The TriggerMesh operator creates and maintains the TriggerMesh serverless management platform on Kubernetes. It provides an easy to use dashboard to Knative with a backend that simplifies functions deployment, event and workflow management.
```

Comment:
It should be mentioned what this Operator requires Knative to be installed. You may also opt for one of the Knative operators as a dependency if you wish.

Reply:
How ? how do you opt-in for the Knative operator ? Knative is definitely a dependency

Reply:
See here: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#required-crds

Depending on other CRDs will cause OLM to resolve dependencies as described in: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/dependency-resolution.md - so if you depend on an CRD of an Operator that is available in any catalogs on, e.g. OpenShift, it will get automatically installed.
This would even allow you to create a Knative deployment using these required CRDs as part of a TriggerMesh deployment. If you want to make this dependency optional, you could ship two CSVs (one with the required CRDs, one without) and server them via different channels as defined in your `package.yaml`

Reply:
@sebgoa The CRD you can depend on is https://github.com/operator-framework/community-operators/blob/master/community-operators/knative-serving-operator/serving_v1alpha1_install_crd.yaml - `installs.serving.knative.dev`. Just depending on that CRD won't actually install Knative for you when your operator is installed, but it will install the Knative Serving Operator.

I'd recommend checking to see if any instances of that `Install` Knative Serving CRD already exist. If so, that means Knative Serving is installed. If not, you can create an instance of that CRD in any namespace and it will install Knative Serving. We do something similar in the Knative Serving operator by ensuring Istio gets installed via its Maistra Operator here - https://github.com/openshift-knative/knative-serving-operator/blob/87439c9f7f562d2d4356a117eec9d934bb6499b0/pkg/controller/install/openshift/openshift.go#L56-L65 . The `Install` resource you create can be as simple as https://gist.githubusercontent.com/bbrowning/8d69e63b0a3b204b123adfb340cb6e1b/raw/90c3aab32abe02cb3f8c0bbd910a8f39b47c99e2/gistfile1.txt and Knative Serving will get installed with all default values.

Reply:
I just mentioned in the description that Knative (serving, build and eventing) is a dependency, since there are no operators for build and eventing yet.